### ISSUE TITLE
NestedScrollView  Looking up a deactivated widget's ancestor is unsafe.

### ISSUE THREAD

**Issue by zmtzawqlp** [flutter/flutter#132190]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

1. run the demo code 
2. drag down on loading screen
3. when future is completed , error will throw.

### Expected results

no error

### Actual results

 error throw

### Code sample

<details><summary>Code sample</summary>

```dart
 import 'package:flutter/cupertino.dart';
import 'package:flutter/material.dart';

void main() {
  runApp(MaterialApp(
    home: RefreshList(),
  ));
}

class RefreshList extends StatefulWidget {
  @override
  State<RefreshList> createState() => _RefreshListState();
}

class _RefreshListState extends State<RefreshList> {
  bool _isLoading = true;
  @override
  void initState() {
    // TODO: implement initState
    super.initState();
    Future<void>.delayed(Duration(seconds: 2)).whenComplete(() {
      setState(() {
        _isLoading = false;
      });
    });
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text('RefreshList'),
      ),
      // AppBar
      body: NestedScrollView(
          headerSliverBuilder: (BuildContext context, bool innerBoxIsScrolled) {
            return [
              const SliverToBoxAdapter(
                child: SizedBox(
                  height: 90.0,
                ),
              ),
            ];
          },
          body: _isLoading
              ? CustomScrollView(
                  physics: BouncingScrollPhysics(),
                  slivers: <Widget>[
                    SliverFillRemaining(
                      child: Center(
                          child: Row(
                        mainAxisAlignment: MainAxisAlignment.center,
                        crossAxisAlignment: CrossAxisAlignment.center,
                        children: <Widget>[
                          Container(
                            margin: const EdgeInsets.only(right: 0.0),
                            height: 30.0,
                            width: 30.0,
                            child: getIndicator(context),
                          ),
                          Text('loading...')
                        ],
                      )),
                    ),
                  ],
                )
              : ListView.builder(itemBuilder: (b, index) {
                  return Container(
                    child: Text('$index'),
                  );
                })),
    );
  }

  Widget getIndicator(BuildContext context) {
    final ThemeData theme = Theme.of(context);
    return theme.platform == TargetPlatform.iOS
        ? const CupertinoActivityIndicator(
            animating: true,
            radius: 16.0,
          )
        : CircularProgressIndicator(
            strokeWidth: 2.0,
            valueColor: AlwaysStoppedAnimation<Color>(theme.primaryColor),
          );
  }
}

```

</details>

### Screenshots or Video

<details>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

https://github.com/flutter/flutter/assets/16477333/39017bf4-baee-4c90-b6f0-7955d115b479

</details>

### Logs

<details><summary>Logs</summary>

```console
════════ Exception caught by widgets library ═══════════════════════════════════
The following assertion was thrown building PrimaryScrollController(no controller):
Looking up a deactivated widget's ancestor is unsafe.

At this point the state of the widget's element tree is no longer stable.

To safely refer to a widget's ancestor in its dispose() method, save a reference to the ancestor by calling dependOnInheritedWidgetOfExactType() in the widget's didChangeDependencies() method.

The relevant error-causing widget was
ListView
main.dart:67
When the exception was thrown, this was the stack
#0      Element._debugCheckStateIsActiveForAncestorLookup.<anonymous closure>
framework.dart:4347
#1      Element._debugCheckStateIsActiveForAncestorLookup
framework.dart:4361
#2      Element.getElementForInheritedWidgetOfExactType
framework.dart:4396
#3      InheritedModel._findModels
inherited_model.dart:143
#4      InheritedModel.inheritFrom
inherited_model.dart:191
#5      MediaQuery._maybeOf
media_query.dart:1058
#6      MediaQuery.maybeDevicePixelRatioOf
media_query.dart:1101
#7      ScrollPosition.devicePixelRatio
scroll_position.dart:244
#8      ScrollMetrics.copyWith
scroll_metrics.dart:57
#9      ScrollPosition.didUpdateScrollDirection
scroll_position.dart:922
#10     _NestedScrollCoordinator.updateUserScrollDirection
nested_scroll_view.dart:634
#11     _NestedScrollPosition.goIdle
nested_scroll_view.dart:1344
#12     new _NestedScrollPosition
nested_scroll_view.dart:1184
#13     _NestedScrollController.createScrollPosition
nested_scroll_view.dart:1119
#14     ScrollableState._updatePosition
scrollable.dart:522
#15     ScrollableState.didChangeDependencies
scrollable.dart:556
#16     StatefulElement._firstBuild
framework.dart:5237
#17     ComponentElement.mount
framework.dart:5062
...     Normal element mounting (13 frames)

```

</details>

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
[✓] Flutter (Channel stable, 3.10.0, on macOS 12.6 21G115 darwin-x64, locale zh-Hans-CN)
    • Flutter version 3.10.0 on channel stable at /Users/zmtzawqlp/Documents/flutter/3.10.0
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 84a1e904f4 (3 months ago), 2023-05-09 07:41:44 -0700
    • Engine revision d44b5a94c9
    • Dart version 3.0.0
    • DevTools version 2.23.1
    • Pub download mirror https://pub.flutter-io.cn
    • Flutter download mirror https://storage.flutter-io.cn

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/zmtzawqlp/Library/Android/sdk
    • Platform android-33, build-tools 30.0.3
    • ANDROID_HOME = /Users/zmtzawqlp/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14C18
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] VS Code (version 1.80.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.70.0

[✓] Connected device (4 available)
    • MI 6 (mobile)              • 5a4b233b                             • android-arm64  • Android 9 (API 28)
    • iPhone 14 Pro Max (mobile) • 12E31781-40FE-4E3D-8D33-40C1A1E49BAD • ios            • com.apple.CoreSimulator.SimRuntime.iOS-16-2 (simulator)
    • macOS (desktop)            • macos                                • darwin-x64     • macOS 12.6 21G115 darwin-x64
    • Chrome (web)               • chrome                               • web-javascript • Google Chrome 115.0.5790.114
```

</details>

**Comment by zmtzawqlp** [comment]
There is no error on v3.3.9, but it happens afer 3.10.0

**Comment by darshankawar** [comment]
@zmtzawqlp 
I tried the code sample using latest stable (3.10.6) running on Android emulator through Android Studio with which I didn't observe the reported error. The list scrolls properly without throwing the error and I get below in the console while scrolling.

```
✓  Built build/app/outputs/flutter-apk/app-debug.apk.
Installing build/app/outputs/flutter-apk/app-debug.apk...
Debug service listening on ws://127.0.0.1:64314/YcL_p03BrB0=/ws
Syncing files to device sdk gphone x86 64...
I/ple.new_project(19868): Compiler allocated 4533KB to compile void android.view.ViewRootImpl.performTraversals()
W/Parcel  (19868): Expecting binder but got null!
D/EGL_emulation(19868): app_time_stats: avg=40.26ms min=6.03ms max=384.35ms count=22
D/EGL_emulation(19868): app_time_stats: avg=369.99ms min=4.04ms max=3695.71ms count=11
D/EGL_emulation(19868): app_time_stats: avg=25.40ms min=3.40ms max=90.20ms count=36
D/EGL_emulation(19868): app_time_stats: avg=162.62ms min=3.55ms max=955.71ms count=6
D/EGL_emulation(19868): app_time_stats: avg=127.87ms min=2.50ms max=1874.05ms count=17
D/EGL_emulation(19868): app_time_stats: avg=511.01ms min=113.46ms max=908.57ms count=2
D/EGL_emulation(19868): app_time_stats: avg=7605.59ms min=7605.59ms max=7605.59ms count=1
D/EGL_emulation(19868): app_time_stats: avg=988.32ms min=3.55ms max=28058.20ms count=29
D/EGL_emulation(19868): app_time_stats: avg=47.70ms min=6.29ms max=1317.73ms count=47
D/EGL_emulation(19868): app_time_stats: avg=1344.82ms min=1344.82ms max=1344.82ms count=1
D/EGL_emulation(19868): app_time_stats: avg=25.32ms min=4.53ms max=162.82ms count=39
D/EGL_emulation(19868): app_time_stats: avg=32.09ms min=9.13ms max=48.07ms count=31
D/EGL_emulation(19868): app_time_stats: avg=99.33ms min=9.38ms max=649.45ms count=10
D/EGL_emulation(19868): app_time_stats: avg=41.11ms min=10.88ms max=175.50ms count=25
D/EGL_emulation(19868): app_time_stats: avg=37.19ms min=6.88ms max=138.23ms count=26
D/EGL_emulation(19868): app_time_stats: avg=213.08ms min=16.48ms max=1132.13ms count=6
D/EGL_emulation(19868): app_time_stats: avg=1333.06ms min=1333.06ms max=1333.06ms count=1
D/EGL_emulation(19868): app_time_stats: avg=18.98ms min=3.73ms max=42.44ms count=43
D/EGL_emulation(19868): app_time_stats: avg=18.78ms min=3.70ms max=140.13ms count=42
D/EGL_emulation(19868): app_time_stats: avg=47.01ms min=4.76ms max=462.19ms count=24
D/EGL_emulation(19868): app_time_stats: avg=29.00ms min=6.35ms max=131.74ms count=34
Application finished.

```

If you could provide more details around it, will be good to properly address the issue.

**Comment by zmtzawqlp** [comment]
> @zmtzawqlp I tried the code sample using latest stable (3.10.6) running on Android emulator through Android Studio with which I didn't observe the reported error. The list scrolls properly without throwing the error and I get below in the console while scrolling.
> 
> ```
> ✓  Built build/app/outputs/flutter-apk/app-debug.apk.
> Installing build/app/outputs/flutter-apk/app-debug.apk...
> Debug service listening on ws://127.0.0.1:64314/YcL_p03BrB0=/ws
> Syncing files to device sdk gphone x86 64...
> I/ple.new_project(19868): Compiler allocated 4533KB to compile void android.view.ViewRootImpl.performTraversals()
> W/Parcel  (19868): Expecting binder but got null!
> D/EGL_emulation(19868): app_time_stats: avg=40.26ms min=6.03ms max=384.35ms count=22
> D/EGL_emulation(19868): app_time_stats: avg=369.99ms min=4.04ms max=3695.71ms count=11
> D/EGL_emulation(19868): app_time_stats: avg=25.40ms min=3.40ms max=90.20ms count=36
> D/EGL_emulation(19868): app_time_stats: avg=162.62ms min=3.55ms max=955.71ms count=6
> D/EGL_emulation(19868): app_time_stats: avg=127.87ms min=2.50ms max=1874.05ms count=17
> D/EGL_emulation(19868): app_time_stats: avg=511.01ms min=113.46ms max=908.57ms count=2
> D/EGL_emulation(19868): app_time_stats: avg=7605.59ms min=7605.59ms max=7605.59ms count=1
> D/EGL_emulation(19868): app_time_stats: avg=988.32ms min=3.55ms max=28058.20ms count=29
> D/EGL_emulation(19868): app_time_stats: avg=47.70ms min=6.29ms max=1317.73ms count=47
> D/EGL_emulation(19868): app_time_stats: avg=1344.82ms min=1344.82ms max=1344.82ms count=1
> D/EGL_emulation(19868): app_time_stats: avg=25.32ms min=4.53ms max=162.82ms count=39
> D/EGL_emulation(19868): app_time_stats: avg=32.09ms min=9.13ms max=48.07ms count=31
> D/EGL_emulation(19868): app_time_stats: avg=99.33ms min=9.38ms max=649.45ms count=10
> D/EGL_emulation(19868): app_time_stats: avg=41.11ms min=10.88ms max=175.50ms count=25
> D/EGL_emulation(19868): app_time_stats: avg=37.19ms min=6.88ms max=138.23ms count=26
> D/EGL_emulation(19868): app_time_stats: avg=213.08ms min=16.48ms max=1132.13ms count=6
> D/EGL_emulation(19868): app_time_stats: avg=1333.06ms min=1333.06ms max=1333.06ms count=1
> D/EGL_emulation(19868): app_time_stats: avg=18.98ms min=3.73ms max=42.44ms count=43
> D/EGL_emulation(19868): app_time_stats: avg=18.78ms min=3.70ms max=140.13ms count=42
> D/EGL_emulation(19868): app_time_stats: avg=47.01ms min=4.76ms max=462.19ms count=24
> D/EGL_emulation(19868): app_time_stats: avg=29.00ms min=6.35ms max=131.74ms count=34
> Application finished.
> ```
> 
> If you could provide more details around it, will be good to properly address the issue.

may be this issue is only on ios, list is apply for BouncingScrollPhysics, which can be drag down

**Comment by zmtzawqlp** [comment]
demo is update, please try again. i change to the physics of CustomScrollView to BouncingScrollPhysics . and i can reproduce it on android device

**Comment by darshankawar** [comment]
@zmtzawqlp 
See https://github.com/flutter/flutter/issues/98104#issuecomment-1034312069 and check if it helps in your case or not.

**Comment by zmtzawqlp** [comment]
> @zmtzawqlp See [#98104 (comment)](https://github.com/flutter/flutter/issues/98104#issuecomment-1034312069) and check if it helps in your case or not.

It looks not the same issue. if the customer is await for the loading page and drag down list, this error will occur，i don't think there is any problem in my code. and Note: This issue did not occur on 3.3.9

**Comment by darshankawar** [comment]
Thanks for the update @zmtzawqlp 
From the video you shared, it seems you were trying to refresh the list even before it was laid properly. Because in that case, the app wouldn't be ready as elements aren't there yet on screen and the tree was still being built. The refresh action would try to schedule a new frame to rebuild the widget tree and hence the error since the UI itself isn't built at first place. If the app elements are rendered and then you try to refresh, does it throw the same error ? 

When I tried the same, it gave me below error:

```
======== Exception caught by foundation library ====================================================
The following assertion was thrown while dispatching notifications for _StretchController:
Build scheduled during frame.

While the widget tree was being built, laid out, and painted, a new frame was scheduled to rebuild the widget tree.

This might be because setState() was called from a layout or paint callback. If a change is needed to the widget tree, it should be applied as the tree is being built. Scheduling a change for the subsequent frame instead results in an interface that lags behind by one frame. If this was done to make your build dependent on a size measured at layout time, consider using a LayoutBuilder, CustomSingleChildLayout, or CustomMultiChildLayout. If, on the other hand, the one frame delay is the desired effect, for example because this is an animation, consider scheduling the frame in a post-frame callback using SchedulerBinding.addPostFrameCallback or using an AnimationController to trigger the animation.

When the exception was thrown, this was the stack: 
#0      WidgetsBinding._handleBuildScheduled.<anonymous closure> (package:flutter/src/widgets/binding.dart:822:9)
#1      WidgetsBinding._handleBuildScheduled (package:flutter/src/widgets/binding.dart:845:6)
#2      BuildOwner.scheduleBuildFor (package:flutter/src/widgets/framework.dart:2707:24)
#3      Element.markNeedsBuild (package:flutter/src/widgets/framework.dart:5036:12)
#4      State.setState (package:flutter/src/widgets/framework.dart:1223:15)
#5      _AnimatedState._handleChange (package:flutter/src/widgets/transitions.dart:129:5)
#6      ChangeNotifier.notifyListeners (package:flutter/src/foundation/change_notifier.dart:403:24)
#7      AnimationLocalListenersMixin.notifyListeners (package:flutter/src/animation/listener_helpers.dart:161:19)
#8      AnimationController.value= (package:flutter/src/animation/animation_controller.dart:363:5)
#9      AnimationController.forward (package:flutter/src/animation/animation_controller.dart:459:7)
#10     _StretchController._recede (package:flutter/src/widgets/overscroll_indicator.dart:938:24)
#11     _StretchController.scrollEnd (package:flutter/src/widgets/overscroll_indicator.dart:911:7)
#12     _StretchingOverscrollIndicatorState._handleScrollNotification (package:flutter/src/widgets/overscroll_indicator.dart:740:26)
#13     _NotificationElement.onNotification (package:flutter/src/widgets/notification_listener.dart:130:38)
#14     _NotificationNode.dispatchNotification (package:flutter/src/widgets/framework.dart:3297:18)
#15     _NotificationNode.dispatchNotification (package:flutter/src/widgets/framework.dart:3300:13)
#16     Element.dispatchNotification (package:flutter/src/widgets/framework.dart:4919:24)
#17     Notification.dispatch (package:flutter/src/widgets/notification_listener.dart:60:13)
#18     ScrollActivity.dispatchScrollEndNotification (package:flutter/src/widgets/scroll_activity.dart:102:63)
#19     ScrollPosition.didEndScroll (package:flutter/src/widgets/scroll_position.dart:1008:15)
#20     ScrollPosition.beginActivity (package:flutter/src/widgets/scroll_position.dart:974:9)
#21     _NestedInnerBallisticScrollActivity.applyNewDimensions (package:flutter/src/widgets/nested_scroll_view.dart:1521:14)
#22     ScrollPosition.applyNewDimensions (package:flutter/src/widgets/scroll_position.dart:706:15)
#23     _NestedScrollPosition.applyNewDimensions (package:flutter/src/widgets/nested_scroll_view.dart:1468:11)
#24     ScrollPosition.applyContentDimensions (package:flutter/src/widgets/scroll_position.dart:635:7)
#25     RenderViewport.performLayout (package:flutter/src/rendering/viewport.dart:1425:20)
#26     RenderObject.layout (package:flutter/src/rendering/object.dart:2548:7)
#27     RenderBox.layout (package:flutter/src/rendering/box.dart:2391:11)
#28     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:104:21)
#29     RenderObject.layout (package:flutter/src/rendering/object.dart:2548:7)
#30     RenderBox.layout (package:flutter/src/rendering/box.dart:2391:11)
#31     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:104:21)
#32     RenderObject.layout (package:flutter/src/rendering/object.dart:2548:7)
#33     RenderBox.layout (package:flutter/src/rendering/box.dart:2391:11)
#34     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:104:21)
#35     RenderObject.layout (package:flutter/src/rendering/object.dart:2548:7)
#36     RenderBox.layout (package:flutter/src/rendering/box.dart:2391:11)
#37     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:104:21)
#38     RenderObject.layout (package:flutter/src/rendering/object.dart:2548:7)
#39     RenderBox.layout (package:flutter/src/rendering/box.dart:2391:11)
#40     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:104:21)
#41     RenderObject.layout (package:flutter/src/rendering/object.dart:2548:7)
#42     RenderBox.layout (package:flutter/src/rendering/box.dart:2391:11)
#43     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:104:21)
#44     RenderObject.layout (package:flutter/src/rendering/object.dart:2548:7)
#45     RenderBox.layout (package:flutter/src/rendering/box.dart:2391:11)
#46     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:104:21)
#47     RenderObject.layout (package:flutter/src/rendering/object.dart:2548:7)
#48     RenderBox.layout (package:flutter/src/rendering/box.dart:2391:11)
#49     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:104:21)
#50     _RenderCustomClip.performLayout (package:flutter/src/rendering/proxy_box.dart:1444:11)
#51     RenderObject.layout (package:flutter/src/rendering/object.dart:2548:7)
#52     RenderBox.layout (package:flutter/src/rendering/box.dart:2391:11)
#53     RenderSliverFillRemainingWithScrollable.performLayout (package:flutter/src/rendering/sliver_fill.dart:89:14)
#54     RenderObject.layout (package:flutter/src/rendering/object.dart:2548:7)
#55     RenderViewportBase.layoutChildSequence (package:flutter/src/rendering/viewport.dart:534:13)
#56     RenderViewport._attemptLayout (package:flutter/src/rendering/viewport.dart:1512:12)
#57     RenderViewport.performLayout (package:flutter/src/rendering/viewport.dart:1421:20)
#58     RenderObject._layoutWithoutResize (package:flutter/src/rendering/object.dart:2387:7)
#59     PipelineOwner.flushLayout (package:flutter/src/rendering/object.dart:1027:18)
#60     PipelineOwner.flushLayout (package:flutter/src/rendering/object.dart:1040:15)
#61     RendererBinding.drawFrame (package:flutter/src/rendering/binding.dart:591:23)
#62     WidgetsBinding.drawFrame (package:flutter/src/widgets/binding.dart:959:13)
#63     RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:457:5)
#64     SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:1325:15)
#65     SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:1255:9)
#66     SchedulerBinding._handleDrawFrame (package:flutter/src/scheduler/binding.dart:1113:5)
#67     _invoke (dart:ui/hooks.dart:170:13)
#68     PlatformDispatcher._drawFrame (dart:ui/platform_dispatcher.dart:410:5)
#69     _drawFrame (dart:ui/hooks.dart:140:31)
The _StretchController sending notification was: _StretchController()
====================================================================================================

```

So see if below message helps to fix the error or not in your code:

`This might be because setState() was called from a layout or paint callback. If a change is needed to the widget tree, it should be applied as the tree is being built. Scheduling a change for the subsequent frame instead results in an interface that lags behind by one frame. If this was done to make your build dependent on a size measured at layout time, consider using a LayoutBuilder, CustomSingleChildLayout, or CustomMultiChildLayout. If, on the other hand, the one frame delay is the desired effect, for example because this is an animation, consider scheduling the frame in a post-frame callback using SchedulerBinding.addPostFrameCallback or using an AnimationController to trigger the animation.`

**Comment by zmtzawqlp** [comment]
> Thanks for the update @zmtzawqlp From the video you shared, it seems you were trying to refresh the list even before it was laid properly. Because in that case, the app wouldn't be ready as elements aren't there yet on screen and the tree was still being built. The refresh action would try to schedule a new frame to rebuild the widget tree and hence the error since the UI itself isn't built at first place. If the app elements are rendered and then you try to refresh, does it throw the same error ?
> 
> When I tried the same, it gave me below error:
> 
> ```
> ======== Exception caught by foundation library ====================================================
> The following assertion was thrown while dispatching notifications for _StretchController:
> Build scheduled during frame.
> 
> While the widget tree was being built, laid out, and painted, a new frame was scheduled to rebuild the widget tree.
> 
> This might be because setState() was called from a layout or paint callback. If a change is needed to the widget tree, it should be applied as the tree is being built. Scheduling a change for the subsequent frame instead results in an interface that lags behind by one frame. If this was done to make your build dependent on a size measured at layout time, consider using a LayoutBuilder, CustomSingleChildLayout, or CustomMultiChildLayout. If, on the other hand, the one frame delay is the desired effect, for example because this is an animation, consider scheduling the frame in a post-frame callback using SchedulerBinding.addPostFrameCallback or using an AnimationController to trigger the animation.
> 
> When the exception was thrown, this was the stack: 
> #0      WidgetsBinding._handleBuildScheduled.<anonymous closure> (package:flutter/src/widgets/binding.dart:822:9)
> #1      WidgetsBinding._handleBuildScheduled (package:flutter/src/widgets/binding.dart:845:6)
> #2      BuildOwner.scheduleBuildFor (package:flutter/src/widgets/framework.dart:2707:24)
> #3      Element.markNeedsBuild (package:flutter/src/widgets/framework.dart:5036:12)
> #4      State.setState (package:flutter/src/widgets/framework.dart:1223:15)
> #5      _AnimatedState._handleChange (package:flutter/src/widgets/transitions.dart:129:5)
> #6      ChangeNotifier.notifyListeners (package:flutter/src/foundation/change_notifier.dart:403:24)
> #7      AnimationLocalListenersMixin.notifyListeners (package:flutter/src/animation/listener_helpers.dart:161:19)
> #8      AnimationController.value= (package:flutter/src/animation/animation_controller.dart:363:5)
> #9      AnimationController.forward (package:flutter/src/animation/animation_controller.dart:459:7)
> #10     _StretchController._recede (package:flutter/src/widgets/overscroll_indicator.dart:938:24)
> #11     _StretchController.scrollEnd (package:flutter/src/widgets/overscroll_indicator.dart:911:7)
> #12     _StretchingOverscrollIndicatorState._handleScrollNotification (package:flutter/src/widgets/overscroll_indicator.dart:740:26)
> #13     _NotificationElement.onNotification (package:flutter/src/widgets/notification_listener.dart:130:38)
> #14     _NotificationNode.dispatchNotification (package:flutter/src/widgets/framework.dart:3297:18)
> #15     _NotificationNode.dispatchNotification (package:flutter/src/widgets/framework.dart:3300:13)
> #16     Element.dispatchNotification (package:flutter/src/widgets/framework.dart:4919:24)
> #17     Notification.dispatch (package:flutter/src/widgets/notification_listener.dart:60:13)
> #18     ScrollActivity.dispatchScrollEndNotification (package:flutter/src/widgets/scroll_activity.dart:102:63)
> #19     ScrollPosition.didEndScroll (package:flutter/src/widgets/scroll_position.dart:1008:15)
> #20     ScrollPosition.beginActivity (package:flutter/src/widgets/scroll_position.dart:974:9)
> #21     _NestedInnerBallisticScrollActivity.applyNewDimensions (package:flutter/src/widgets/nested_scroll_view.dart:1521:14)
> #22     ScrollPosition.applyNewDimensions (package:flutter/src/widgets/scroll_position.dart:706:15)
> #23     _NestedScrollPosition.applyNewDimensions (package:flutter/src/widgets/nested_scroll_view.dart:1468:11)
> #24     ScrollPosition.applyContentDimensions (package:flutter/src/widgets/scroll_position.dart:635:7)
> #25     RenderViewport.performLayout (package:flutter/src/rendering/viewport.dart:1425:20)
> #26     RenderObject.layout (package:flutter/src/rendering/object.dart:2548:7)
> #27     RenderBox.layout (package:flutter/src/rendering/box.dart:2391:11)
> #28     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:104:21)
> #29     RenderObject.layout (package:flutter/src/rendering/object.dart:2548:7)
> #30     RenderBox.layout (package:flutter/src/rendering/box.dart:2391:11)
> #31     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:104:21)
> #32     RenderObject.layout (package:flutter/src/rendering/object.dart:2548:7)
> #33     RenderBox.layout (package:flutter/src/rendering/box.dart:2391:11)
> #34     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:104:21)
> #35     RenderObject.layout (package:flutter/src/rendering/object.dart:2548:7)
> #36     RenderBox.layout (package:flutter/src/rendering/box.dart:2391:11)
> #37     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:104:21)
> #38     RenderObject.layout (package:flutter/src/rendering/object.dart:2548:7)
> #39     RenderBox.layout (package:flutter/src/rendering/box.dart:2391:11)
> #40     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:104:21)
> #41     RenderObject.layout (package:flutter/src/rendering/object.dart:2548:7)
> #42     RenderBox.layout (package:flutter/src/rendering/box.dart:2391:11)
> #43     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:104:21)
> #44     RenderObject.layout (package:flutter/src/rendering/object.dart:2548:7)
> #45     RenderBox.layout (package:flutter/src/rendering/box.dart:2391:11)
> #46     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:104:21)
> #47     RenderObject.layout (package:flutter/src/rendering/object.dart:2548:7)
> #48     RenderBox.layout (package:flutter/src/rendering/box.dart:2391:11)
> #49     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:104:21)
> #50     _RenderCustomClip.performLayout (package:flutter/src/rendering/proxy_box.dart:1444:11)
> #51     RenderObject.layout (package:flutter/src/rendering/object.dart:2548:7)
> #52     RenderBox.layout (package:flutter/src/rendering/box.dart:2391:11)
> #53     RenderSliverFillRemainingWithScrollable.performLayout (package:flutter/src/rendering/sliver_fill.dart:89:14)
> #54     RenderObject.layout (package:flutter/src/rendering/object.dart:2548:7)
> #55     RenderViewportBase.layoutChildSequence (package:flutter/src/rendering/viewport.dart:534:13)
> #56     RenderViewport._attemptLayout (package:flutter/src/rendering/viewport.dart:1512:12)
> #57     RenderViewport.performLayout (package:flutter/src/rendering/viewport.dart:1421:20)
> #58     RenderObject._layoutWithoutResize (package:flutter/src/rendering/object.dart:2387:7)
> #59     PipelineOwner.flushLayout (package:flutter/src/rendering/object.dart:1027:18)
> #60     PipelineOwner.flushLayout (package:flutter/src/rendering/object.dart:1040:15)
> #61     RendererBinding.drawFrame (package:flutter/src/rendering/binding.dart:591:23)
> #62     WidgetsBinding.drawFrame (package:flutter/src/widgets/binding.dart:959:13)
> #63     RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:457:5)
> #64     SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:1325:15)
> #65     SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:1255:9)
> #66     SchedulerBinding._handleDrawFrame (package:flutter/src/scheduler/binding.dart:1113:5)
> #67     _invoke (dart:ui/hooks.dart:170:13)
> #68     PlatformDispatcher._drawFrame (dart:ui/platform_dispatcher.dart:410:5)
> #69     _drawFrame (dart:ui/hooks.dart:140:31)
> The _StretchController sending notification was: _StretchController()
> ====================================================================================================
> ```
> 
> So see if below message helps to fix the error or not in your code:
> 
> `This might be because setState() was called from a layout or paint callback. If a change is needed to the widget tree, it should be applied as the tree is being built. Scheduling a change for the subsequent frame instead results in an interface that lags behind by one frame. If this was done to make your build dependent on a size measured at layout time, consider using a LayoutBuilder, CustomSingleChildLayout, or CustomMultiChildLayout. If, on the other hand, the one frame delay is the desired effect, for example because this is an animation, consider scheduling the frame in a post-frame callback using SchedulerBinding.addPostFrameCallback or using an AnimationController to trigger the animation.`

1. i didn't meet the error which you meantion. you can try to drag down and hold it before _isLoading is true, and after 2 seconds, you will see the error.
2. setState in next frame is not helpful for my issue.

```
  bool _isLoading = true;
  @override
  void initState() {
    // TODO: implement initState
    super.initState();
    Future<void>.delayed(Duration(seconds: 2)).whenComplete(() {
      WidgetsBinding.instance.addPostFrameCallback((t) {
        setState(() {
          _isLoading = false;
        });
      });
    });
  }
```

**Comment by darshankawar** [comment]
Thanks for the update. I am unable to replicate the exact error reported in this issue, but based on it and discussion above, keeping it open for team's attention and tracking.

**Comment by goderbauer** [comment]
cc @Piinks

**Comment by Piinks** [comment]
> I am unable to replicate the exact error reported in this issue

Looks like this was fixed in #126535 which was first reported in #126454  🎉 
The fix should be in the next stable release, but is also currently available on the beta channel.
